### PR TITLE
Using hovering explanations as legends

### PR DESCRIPTION
**Pull request by zhongh**
@kolans @xgmachina Here we use hover explanations as legends for labels carrying mindat-registry, remote dated locality, and detail age informations. Comments and advices?
screen shot 2016-05-02 at 3 51 38 pm
screen shot 2016-05-02 at 3 52 00 pm
screen shot 2016-05-02 at 4 00 04 pm

### CODE REVIEW

**Comment by kolans** [comment]
I like the hover function. I think that looks great! I would suggest having a legend somewhere eventually (maybe as a separate link?) as an index for all the different symbols in case a user sees there are different dating methods and wants to check out what the rest of them are (maybe what they're looking at only has a couple types?). But I'm a fan of just using the hover method on this page.  My other suggestion is that in the hover box, we have "letters: explanation". So it would say, "MD: Mindat claims this mineral occurs at this locality." only because of cases such as the middle hover box  with the remote dated locality where the two symbols are so close together. Offers a kind of assurance to the user they are receiving the information they intend to. Does that make sense?

**Comment by zednis** [comment]
@kolans I think that suggestion makes sense.  Let's update the hover text to be "letters: explanation"

**Comment by zhongh** [comment]
@zednis @kolans "LETTER: Explanation" style:
screen shot 2016-05-04 at 11 38 30 am
screen shot 2016-05-04 at 11 37 55 am
screen shot 2016-05-04 at 11 38 05 am

**Comment by zhongh** [comment]
@kolans @zednis Please check it here: https://deeptime.tw.rpi.edu/map/browser1.html
This one has both the 50 Ma intervals and the hover helpers. Note it's "browser1" not "browser" (just another temporary file in the same directory having both changes)

**Comment by zednis** [comment]
Looks good to me :+1:

**Comment by zednis** [comment]
@zhonghao1014 did you email the keck mailinglist with a pointer to browser1?

**Comment by zhongh** [comment]
@zednis It seems that no one in the KECK mailing list has responded with any suggestions on the updates on age intervals and the hovers.

**Comment by zednis** [comment]
Then let's add this change to the map pop-up, then close the pull request and update browser.html and map.html and remove browser1.html

File: html/browser.html [hunk 1]
```diff
@@ -138,7 +154,7 @@
 
         $(document).ready(function($) {
             $('.dtdi-map').facetview({
-                search_url: 'http://localhost:9200/dtdi/mineral-observation/_search',
+                search_url: 'https://deeptime.tw.rpi.edu/es/dtdi/mineral-observation/_search',
```
Comment (zednis) [inline html/browser.html:157]: remove the https from the url and have it just start with '//'

Comment (zednis) [inline html/browser.html:157]: and we can keep this as localhost

File: html/browser.html [hunk 2]
```diff
@@ -78,7 +68,8 @@
 
                 {{#if mineral-species.rruff-chemistry-html}}
                     <div>
-                        <strong>Mineral Chemistry:</strong>{{{mineral-species.rruff-chemistry-html}}}
+                        <strong>Mineral Chemistry:</strong>
+                        {{{mineral-species.rruff-chemistry-html}}}
```
Comment (zednis) [inline html/browser.html:72]: Why is this in 3 brackets? '{{{' instead of '{{'?

**Comment by zhongh** [comment]
@zednis Please review the hovers on
https://deeptime.tw.rpi.edu/map/browser.html
and
https://deeptime.tw.rpi.edu/map/map.html

**Comment by zednis** [comment]
Looks good.  :shipit: 

note - it took the map a very long time to load when I was checking it, but the browser loaded pretty quickly.  I am not sure if that was random or if it had anything to do with this update.

Let's keep an eye on that.

**Comment by zhongh** [comment]
@zednis In development I noticed that it is a little bit slower than before too. But I was thinking it was trivial. However, since you mentioned it, I thought about it again and realised that it does has something to do with this update. For the browser, each time it is loaded only 20 results are loaded, thus loading the hover contents takes no time; however for the map, each time the page is loaded, all hover contents are loaded at once, which could cause a, not long, but short, noticeable delay.

I think we can improve this issue by loading corresponding hover content only when a point on the map is clicked on.

**Comment by zednis** [comment]
I think that is a good idea, let's try it out before merging.

**Comment by zhongh** [comment]
@zednis I examined the approx. page loading time before this change and after, I found that they both takes like 1-2 seconds to load the page. In fact, the delay should not be due to loading the hover contents of 3000+ points at the same time, but due to loading all the html of 3000+ points at this same time, while the hover content is only a small addition to the whole scale.
I suggest we can ship this branch and get the wheel moving?

**Comment by zednis** [comment]
perhaps we can delay generating the HTML for all 3000+ locations?  The map is taking ~56 seconds to load for me.  Perhaps it is the new badges?

**Comment by zhongh** [comment]
@zednis As I physically locates in IL now, I tried to load the map from here with Safari, it does take a long time, ~30 seconds. I will make changes to delay generating HTML for all 3000+ locations.

**Comment by zednis** [comment]
thanks.  I generally have very fast internet so I found it odd that we were getting such disparate performance when loading the map.

...actually I think I know what is causing the problem.  The badges are SVGs loaded from a service.  For each map location popup (not the new hover) there are 3 badges, so ~9000 SVGs are loaded from the badge service.

**Comment by zednis** [comment]
It would be ideal if we can generate the HTML for the map pop-ups on demand, as that would cover the badge load + all other HTML.  If that is not something we can easily do, let's attempt to just have the badges load on demand.

**Comment by zednis** [comment]
and since we think this is badge or HTML related, let's continue it in a new ticket.

edit: I created #53 for continuing the performance discussion.

**Comment by zednis** [comment]
Let's close this issue and keep working on performance issues in #53

**Comment by zednis** [comment]
:shipit:

**Pull request merged**